spec/javascripts/controllers/schedule/schedule_form_controller_spec.js

```diff
@@ -715,4 +715,43 @@ describe('scheduleFormController', function() {
       });
     });
   });
+
+  describe('saveable should exist in the scope', function() {
+    it('returns true', function() {
+      expect($scope.saveable).toBeDefined();
+    });
+  });
+
+  describe('Validates credential fields', function() {
+    beforeEach(inject(function($compile, miqService) {
+      var angularForm;
+      var element = angular.element(
+        '<form name="angularForm">' +
+        '<input ng-model="scheduleModel.depot_name" name="depot_name" ng-required="scheduleModel.depot_name.length==0" text />' +
+        '<input ng-model="scheduleModel.uri" name="uri" ng-required="scheduleModel.uri.length==0" text />' +
```

Comment:
`ng-required="scheduleModel.uri.length==0"`, @AparnaKarve this is almost certainly wrong, or at least, I see no point in making the field required when it contains something and not required when it's empty. You probably just want `required`?